cli/template_render.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/prometheus/alertmanager/template"
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
+)
+
+type templateRenderCmd struct {
+	templateFilesGlobs []string
+	templateType       string
+	templateText       string
+}
+
+func configureTemplateRenderCmd(cc *kingpin.CmdClause) {
+	var (
+		c         = &templateRenderCmd{}
+		renderCmd = cc.Command("render", "Render a given definition in a template file to standard output")
+	)
+
+	renderCmd.Flag("templateglob", "Glob of paths that will be expanded and used for rendering").Required().StringsVar(&c.templateFilesGlobs)
+	renderCmd.Flag("templatetext", "The template that will be rendered").Required().StringVar(&c.templateText)
+	renderCmd.Flag("templatetype", "The type of the template. Can be either text (default) or html").EnumVar(&c.templateType, "html", "text")
+
+	renderCmd.Action(execWithTimeout(c.render))
+}
+
+func (c *templateRenderCmd) render(ctx context.Context, _ *kingpin.ParseContext) error {
+	tmpl, err := template.FromGlobs(c.templateFilesGlobs...)
+	if err != nil {
+		return err
+	}
+
+	f := tmpl.ExecuteTextString
+	if c.templateType == "html" {
+		f = tmpl.ExecuteHTMLString
+	}
+
+	data := template.Data{
```

Comment:
We might want to load this from a file or something so that users could try out how their logic works with certain labels/annotations. WDYT?

Reply:
I think the ability to load custom data from YAML or JSON would be ideal, like in https://juliusv.com/promslack/.

Reply:
Looks good to me ;)